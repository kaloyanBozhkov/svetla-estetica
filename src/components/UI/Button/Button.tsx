import type { ComponentPropsWithoutRef, ReactElement, ReactNode } from 'react'
import styles from './button.module.scss'

interface ButtonProps extends ComponentPropsWithoutRef<'button'> {
    label: ReactNode
    modifier?: 'hallow' | 'solid' | 'denied'
    disabled?: boolean
}

const Button = ({
    label,
    onClick,
    modifier = 'solid',
    disabled = false,
    type = 'button',
}: ButtonProps): ReactElement => (
    <button
        className={styles.button}
        onClick={onClick}
        data-modifier={modifier}
        disabled={disabled}
        // eslint-disable-next-line
        type={type}
    >
        {label}
    </button>
)

export default Button
