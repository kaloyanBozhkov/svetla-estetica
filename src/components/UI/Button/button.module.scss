@import "scss/variables.scss";

.button {
    width: 100%;
    height: 100%;

    &[data-modifier="hallow"] {
        border: 1px solid $white;
        border-radius: 3px;
        color: $white;
        transition: all 300ms;
        &:hover {
            background: $white;
        }
    }
    &[data-modifier="solid"] {
        border: 1px solid $lighterGrey;
        border-radius: 3px;
        color: $black;
        background-color: white;
        transition: all 300ms;
        &:hover {
            background: $lighterGrey;
        }
    }
    &[data-modifier="denied"] {
        border: 1px solid $pink;
        border-radius: 3px;
        color: $black;
        background-color: white;
        transition: all 300ms;
        &:hover {
            background: $pink;
        }
    }

    &[disabled] {
        cursor: not-allowed;
    }
}
