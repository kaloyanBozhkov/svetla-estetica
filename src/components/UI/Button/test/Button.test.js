import React from 'react'
import { shallow } from 'enzyme'
import Button from '../Button'

describe('testing BUtton', () => {
    it('should render atom', () => {
        expect(
            shallow(<Button label="Some Label" onClick={jest.fn()} modifier="hallow" />)
        ).toMatchSnapshot()
    })

    it('should fire onClick fn', () => {
        const mockClick = jest.fn(),
            wrapper = shallow(<Button label="Some Label" onClick={mockClick} modifier="hallow" />)
        wrapper.prop('onClick')()
        expect(mockClick).toHaveBeenCalled()
    })
})
