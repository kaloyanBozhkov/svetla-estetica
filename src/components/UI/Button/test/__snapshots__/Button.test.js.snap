// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing BUtton should render atom 1`] = `
<button
  className="button"
  data-modifier="hallow"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Some Label
</button>
`;
